### ISSUE TITLE
Model Contain Text and numeric features 0.8 issue

### ISSUE THREAD

**Issue by DevLob-zz** [dotnet/machinelearning#1987]
### System information

- **OS version/distro**:W10
- **.NET Version (eg., dotnet --info)**: 4.61

### Issue

- **What did you do?**i read training data from database then convert to to IDataView i try to learn the Model 
My Model Object Contain numeric value + Text values 

i used 0.8 version
- **What happened?** throw exception 'Column 'tagText' has values of Textwhich is not the same as earlier observed type of R4.'
- **What did you expect?** should be worked as schema read correctly [observed type of R4. i don't know when this value read from]

### Source code / logs

Please paste or attach the code or logs or traces that would be helpful to diagnose the issue you are reporting.

`public class NormalTagsModelFeatures
    {
        [Column(ordinal: "0", name: "Label")] public string Label;

        [Column("1")] public float fontSize;
        [Column("2")] public float isBold;
        [Column("3")] public float isItalic;
        [Column("4")] public float isUnderLine;
        [Column("5")] public float containsDot;
        [Column("6")] public float containsQuestionMark;
        [Column("7")] public string fontColor;
        [Column("8")] public float isAllCaps;
        [Column("9")] public string tagText;
        [Column("10")] public string firstWord;

    }`

`    public class NormalTagsPrediction
    {
        [Column(ordinal: "0", name: "PredictedLabel")]
        // public string Class;
        public string Label;
        public float[] Score { get; set; }
    }`

`public  void BuildNormalTrainEvaluateAndSaveModel()
        {
            //Set a random seed for repeatable/deterministic results across multiple trainings.
            var mlContext = new MLContext(seed: 1);

            #region "STEP 1: Common data loading configuration"
            IDataView trainingDataView = GetNormalDataSet(mlContext);
            
            //var env = mlContext.Data.GetEnvironment();
            //IDataView trainingDataView  =mlContext.Data.ReadFromBinary(stream);
            #endregion
            #region "STEP 2: Common data process configuration with pipeline data transformations"
         
            var dataProcessPipeline = mlContext.Transforms.Concatenate("Features", "fontSize",
                                                                                       "isBold",
                                                                                       "isItalic",
                                                                                       "isUnderLine",
                                                                                       "containsDot",
                                                                                       "containsQuestionMark",
                                                                                       "isAllCaps",
                                                                                        //"pageNo",
                                                                                        //"tagHeight",
                                                                                        //"tagWidth",
                                                                                        //"NumericFeatures"
                                                                                        //"fontColor",
                                                                                        "tagText",
                                                                                        "firstWord");
            var dataProcessPipeline1 = mlContext.Transforms.Text.FeaturizeText(new List<string> { "tagText", "firstWord" }, "Features");
            

            //mlContext.Transforms.CustomMapping()
            #endregion
            #region  "STEP 3: Set the training algorithm, then create and config the modelBuilder"                            
            var modelBuilder = new ModelBuilder<NormalTagsModelFeatures, NormalTagsPrediction>(mlContext, dataProcessPipeline);
            // We apply our selected Trainer 
            
            var trainer = mlContext.MulticlassClassification.Trainers.StochasticDualCoordinateAscent(labelColumn: "Label", featureColumn: "Features");
            
            modelBuilder.AddTrainer(trainer);
            #endregion
            #region "STEP 4: Train the model fitting to the DataSet"
            //The pipeline is trained on the dataset that has been loaded and transformed.
            //Console.WriteLine("=============== Training the model ===============");
           
            modelBuilder.Train(trainingDataView);
            #endregion
            #region "STEP 5: Evaluate the model and show accuracy stats"
            //Console.WriteLine("===== Evaluating Model's accuracy with Test data =====");
            var metrics = modelBuilder.EvaluateMultiClassClassificationModel(trainingDataView, "Label");
            Common.ConsoleHelper.PrintMultiClassClassificationMetrics(trainer.ToString(), metrics);
            #endregion
            #region "STEP 6: Save/persist the trained model to a .ZIP file
            //Console.WriteLine("=============== Saving the model to a file ===============");
            string fullFilePath = GetModelPath("Norm");
            UpdateNormalModelPath(fullFilePath);
            modelBuilder.SaveModelAsFile(fullFilePath);
            #endregion
        }`

![image](https://user-images.githubusercontent.com/5037612/50536401-b4d0f280-0b5c-11e9-856b-3b0cafd1a9be.png)

![image](https://user-images.githubusercontent.com/5037612/50536406-c3b7a500-0b5c-11e9-8c79-2902bed261a5.png)

**Comment by yaeldekel** [comment]
Hi @DevLob-zz , thank you for your question! The reason you are seeing this error is because in your definition of dataProcessPipeline you are trying to concatenate numeric columns (such as fontSize and isBold) with text columns (such as tagText and firstWord). The solution to this, is to apply the text featurization before you concatenate, i.e., define dataProcessPipeline1 before you define dataProcessPipeline, and then define dataProcessPipeline this way:

var dataProcessPipeline = dataProcessPipeline1.Append(...); (with the definition of the mlContext.Transforms.Concatenate inside).

Here is an example of using Append to create a multi step data processing pipeline:

https://github.com/dotnet/machinelearning-samples/blob/master/samples/csharp/getting-started/Regression_TaxiFarePrediction/TaxiFarePrediction/TaxiFarePredictionConsoleApp/Program.cs#L78
